### PR TITLE
Vendor updates for the Go codegen for KFservice

### LINKED ISSUE #161: Proposal: Add Go Client Libraries fo KFservice
/kind feature

Should we autogen client libraries for our types in Go?
This will allow users of KfServing to easily manipulate kfservices.

There seems to be an open issue for this on kubebuilder https://github.com/kubernetes-sigs/kubebuilder/issues/403 which suggests as the k8s Go client has dynamic-client then this is not required. 

It still seems to me to have static exposed client informers etc would still be useful and easy to add/update as per other libraries such as knative do.

### PR DESCRIPTION
**Pull request by SachinVarghese**
**What this PR does / why we need it**:
This PR pertains to the vendor updates for the go-client apis to be created for KFservice CRD using the kubernetes code-generator project. 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #161
Related to PR https://github.com/kubeflow/kfserving/pull/177

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kfserving/197)

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -22,6 +22,15 @@ required = [
...
   non-go = true
 
+[[prune.project]]
+  name = "k8s.io/code-generator"
+  non-go = false
+  unused-packages = false
+
+[[prune.project]]
+  name = "k8s.io/gengo"
+  unused-packages = false
```
Comment (yuzisun) [inline Gopkg.toml]: Looks the added files from gengo are all from examples directory, are we using them?

Comment (SachinVarghese) [reply Gopkg.toml]: Yeah we actually need this package k8s.io/gengo/examples/defaulter-gen/generators

**Comment by yuzisun** [comment]
Thanks @SachinVarghese !
/lgtm
/approve

**Pull request merged**